packages/components/src/navigator/types.ts

```diff
@@ -12,11 +12,11 @@ export type MatchParams = Record< string, string | string[] >;
 
 type NavigateOptions = {
 	focusTargetSelector?: string;
+	isBack?: boolean;
```

Comment:
Yeah I'm aware of it and I actually didn't see any harm. (It kind of enables the third option I proposed implicitly).  There's of course a possibility to hide it by having "two" go to functions. one public and one private but it felt unnecessary to me.

Reply:
I definitely see the flexibility in keeping the `isBack` option available in `goTo` — as you say, consumers of `Navigator` could easily just use `goTo` and ignore `goBack` / `goToParent`.

On top of my previous worries, we should also consider that a consumer only using `goTo` would cause the `locationHistory` array to grow indefinitely, potentially causing a lot of memory usage.

I guess a potential compromise here could be:
- move the short circuit to `goBack()` from `goToParent` to `goTo`, so that we can call `goBack` wherever possible even when `goTo` is called
- potentially add a word of caution in the docs about memory usage?

Reply:
I always thought the location history should have a limit, so maybe instead we could put a max length to the history (like say 10 items)

Reply:
To be clear, I don't mind the other options, just sharing another one.

Reply:
Alright, what about:

- we keep the `isBack` option available in `goTo`
- we move the "short circuit" to `goBack` from `goToParent` to `goTo`, so that we can internally use `goBack` as much as possible
- we add a max size to `locationHistory`. I think we could start with a high number (say 50?) and see how it goes? For the implementation, I was thinking: when the array is full, `push`ing one item will result in the array being `shift`ed, meaning that the oldest (first) item gets removed.